Draft-Accepted/RFCNNNN-Install-PowerShell.md

```diff
@@ -0,0 +1,177 @@
+ versions. So even if size was not an issue, we cannot check in a version of PS7 with a .NET that is
+ not supported while Windows itself is supported.
+
+### Solution
+
+To solve these issues, the proposal is to ship a new command in Windows and the PowerShell Gallery
+that would:
+
+- Be locally discoverable
+- Perform a simple download and install of the latest LTS release of PowerShell
+- Include parameter options to select Stable, Preview, or LTS releases
+
+## Goals/Non-Goals
+
+Goals:
+
+- By default, download and install the latest LTS release to the users CurrentUser Scope
+  - CurrentUser - default installation scope location $env:LOCALAPPDATA\Microsoft\pwsh
+  - AllUsers - Administrative users may explicit select this scope. AllUsers requires administrative
+    elevation and performs a silent install using the MSI defaults.
+- Allow users to use remoting for install at scale with Invoke-Command
+- Allow advanced users to specify channel Stable/Preview/LTS
+
+Non-goals:
+
+- Removal of PowerShell (Remove-PowerShell7)
+- Updating of PowerShell (Update-PowerShell7)
+- User selected versions prior to current (No version parameter)
+- Disconnected scenarios - Users that are air-gapped may download the appropriate release from GitHub.
+- Handling of additional MSI switch options - i.e. Enable PSRemoting
+- Cross platform support
+
+## Alternatives
+
+`install-powershell7` cmdlet is an alternative approach that is more common to discover for a
+Powershell user. The cmdlet approach increases the complexity of installing a new version of
+PowerShell from an existing version. Additionally, The ability to bootstrap and install PowerShell
+on a system that currently does not include Powershell is an added benefit for using a command.
+
+`install-powershell7` with -Scope CurrentUser installs PowerShell to $env:LOCALAPPDATA\Microsoft\pwsh. The user may wish to specify a location for the installation with a parameter **Path**.
+
+## Specification
+
+Command name: install-powershell7
+
+The default behavior is to install to CurrentUser scope, with the latest LTS
+release of PowerShell. If the user has administrative
+privilege, they can choose to install to the scope of AllUsers.
+
+- The package install location - $env:LOCALAPPDATA\Microsoft\pwsh
+- The installer should update the Users PATH for pwsh
+- The installer should enable Microsoft Updates for scope AllUsers. 
+
+The following parameters can be added by experienced PowerShell users to customize the installation.
+
+- Parameter: Scope - overrides the default of CurrentUser and allows the user to specify the
+  installation location AllUsers.
+- Parameter: Channel - options Stable/Preview/LTS - allows the user to select if they want the
+  current preview or LTS release.
+
+## Syntax
+
+The command `install-powershell7` is lower case and supports `/` switch options to be consistent
+with other command-line tools.
```

Comment:
> supports `/` switch options to be consistent with other command-line tools.

I think we need to support the `-` and `--` switch options as it's more of a standard for command-line tools.
We can also support `/` switch options to be consistent with legacy Windows commands.
For example, `wt` (Windows Terminal) supports `-/--` and `/`, but it's help content only shows the `-/--` options (try `wt --help`). 
`wsl` (WSL command-line tool) supports `-/--` options only.

Reply:
We talked about this and decided that was an implementation detail up to the developer.

The main idea is that we didn't want parameters with arguments. To keep it simple, it only uses mutually-exclusive option switches.

Reply:
I think what form of options to support should be called out in the RFC. `-/--` options are obviously more consistent with modern command-line tools and should be the default.

The RFC should decide whether we want to support the `/` form or not. They are mainly for consistency with legacy Windows command-line tools. If it's up to the developer to decide, then it's likely `/` won't be supported because that's probably extra work.

Reply:
Thank you - I will change to -/--